deepmd/train/trainer.py

```diff
@@ -345,6 +358,15 @@ def _build_training(self):
             optimizer = self.run_opt._HVD.DistributedOptimizer(optimizer)
         else:
             optimizer = tf.train.AdamOptimizer(learning_rate = self.learning_rate)
+        if self.mixed_prec is not None:
+            TF_VERSION_LIST = [int(item) for item in TF_VERSION.split('.')]
```

Comment:
`int(item)` will cause an error if the version is a pre-release, e.g. `v2.6.0-rc1`. See https://github.com/tensorflow/tensorflow/blob/ff68385595088304cf772086b9a259a65b007622/tensorflow/core/public/version.h#L35-L37

Reply:
I suggest to use a third-party class [Specifiers](https://packaging.pypa.io/en/latest/specifiers.html)